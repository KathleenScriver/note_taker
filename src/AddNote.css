.add-note {
  margin: 20px;
  padding: 5px;
  border: 2px solid #00FFD8;
  background: #DDFFEE;
  text-align: center;
  border-radius: 20px;
  box-shadow: 5px 10px 10px #888888;
  padding-bottom: 50px;
}

.input-area {
  align-items: center;
}

.input-area textarea {
  margin-top: 1px;
  margin-bottom: 5px;
  height: 120px;
  width: 220px;
  border-radius: 5px;
}

.button-section {
  display: flex;
  justify-content: space-around;
  align-items: center;
  flex-flow: row || nowrap;
}

.tag-list {
  height: 30px;
  width: 100px;
  font-size: 15px;
}

.add-button {
  margin: 10px;
  margin-top: 20px;
  background-color: #769FB6;
  border-color: #538083;
  border-width: 1px;
  border-radius: 10px;
  padding: 5px;
  color: white;
  width: 100px;
  text-align: center;
  font-size: 14px;
}
