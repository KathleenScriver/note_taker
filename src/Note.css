.note {
  border: 2px solid black;
  text-align: center;
  background: #f5f5f5;
  color: #333;
  margin: 20px;
  padding: 20px;
  text-transform: capitalize;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.text {
  flex-basis: 50%;
}

.remove {
  background-color: #9AD4D6;
  border-color: #538083;
  border-width: 1px;
  border-radius: 10px;
}
