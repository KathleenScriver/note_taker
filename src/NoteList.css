.note-list {
  margin: 20px;
  padding: 5px;
  border: 2px solid #00D8FF;
  background: #DDEEFF;
  border-radius: 20px;
  box-shadow: 5px 10px 10px #888888;
}
