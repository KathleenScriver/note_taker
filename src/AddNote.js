import React, { Component } from 'react';
import './AddNote.css';

export default class AddNote extends Component {
  constructor(props) {
    super(props);
    this.state = {
      noteText: ''
    }
  }

  updateNote = (event) => {
    this.setState({ noteText: event.target.value });
  }

  addNote = () => {
    this.props.addNote(this.state.noteText);
    this.setState({ noteText: '' });
  }

  render() {
    return (
      <div className="add-note">
        <h1>Add a Note</h1>
        <textarea
          type='text'
          max='250'
          onChange={this.updateNote}
          value={this.state.noteText}
        />
        <button onClick={this.addNote}>Add Note</button>
      </div>
    )
  }
}
