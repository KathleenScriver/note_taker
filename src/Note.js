import React, { Component } from 'react';
import './Note.css';

export default class Note extends Component {
  removeNote = () => {
    this.props.removeNote(this.props.name);
  }

  render() {
    return (
      <div className='note'>
        <p>{this.props.name} || {this.props.tag}</p>
        <button className='remove' onClick={this.removeNote}>X</button>
      </div>
    )
  }
}
