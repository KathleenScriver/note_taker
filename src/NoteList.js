import React, { Component } from 'react';
import './NoteList.css';
import Note from './Note'

export default class NoteList extends Component {
  constructor(props) {
    super(props);
    this.state = {
      notes: ['Get groceries', 'Finish project']
    }
  }

  renderNotes = () => {
    return this.state.notes.map(note => (
      <Note name={note} />
    ));
  }

  render() {
    return (
      <div className='note-list'>
        {this.renderNotes()}
      </div>
    )
  }
}
