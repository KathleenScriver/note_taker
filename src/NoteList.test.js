import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import NoteList from './NoteList';
import AddNote from './AddNote';
import Note from './Note';

describe(NoteList, () => {
  const component = shallow(
    <NoteList />
  );

  it('renders and matches our snapshot', () => {
    const component = renderer.create(
      <NoteList />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('contains an AddNote subcomponent', () => {
    expect(component.find(AddNote)).toHaveLength(1);
  })

  it('contains same number of Note components as notes', () => {
    const notesComponents = component.find(Note).length;
    const notes = component.state('notes').length;
    expect(notesComponents).toEqual(notes);
  });

  it('adds note when add note function called', () => {
    const before = component.find(Note).length;
    component.instance().addNote('Test');
    component.update();
    const after = component.find(Note).length;
    expect(after).toBeGreaterThan(before);
  });

  it('removes note when remove note function called', () => {
    const before = component.find(Note).length;
    const removeThisOne = component.state('notes')[0];
    component.instance().removeNote(removeThisOne);
    component.update();
    const after = component.find(Note).length;
    expect(after).toBeLessThan(before);
  })
})
