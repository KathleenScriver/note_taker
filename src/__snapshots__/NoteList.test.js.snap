// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteList renders and matches our snapshot 1`] = `
<div
  className="main-note-display"
>
  <div
    className="add-note"
  >
    <h1>
      Add a Note
    </h1>
    <textarea
      max="250"
      onChange={[Function]}
      type="text"
      value=""
    />
    <select
      className="tag-list"
      onChange={[Function]}
      required={true}
    >
      <option
        default={true}
        value=""
      >
        Select Tag
      </option>
      <option
        value="hobby"
      >
        Hobby
      </option>
      <option
        value="work"
      >
        Work
      </option>
      <option
        value="personal"
      >
        Personal
      </option>
    </select>
    <button
      onClick={[Function]}
    >
      Add Note
    </button>
  </div>
  <div
    className="note-list"
  >
    <h1>
      Notes
    </h1>
    <p>
      "Loading..."
    </p>
  </div>
</div>
`;
